fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownJoinPredicateBase.java

```diff
@@ -107,16 +106,12 @@ public static OptExpression pushDownOnPredicate(OptExpression input, ScalarOpera
 
         OptExpression root;
         if (joinPredicate == null) {
-            if (join.getJoinType().isInnerJoin() || join.getJoinType().isCrossJoin()) {
-                LogicalJoinOperator crossJoin = new LogicalJoinOperator.Builder().withOperator(join)
-                        .setJoinType(JoinOperator.CROSS_JOIN)
-                        .setOnPredicate(null)
-                        .setPredicate(postJoinPredicate)
-                        .build();
-                root = OptExpression.create(crossJoin, input.getInputs());
-            } else {
-                throw new SemanticException("No equal on predicate in " + join.getJoinType() + " is not supported");
-            }
+            LogicalJoinOperator crossJoin = new LogicalJoinOperator.Builder().withOperator(join)
```

Comment:
why remove this check?

Reply:
The purpose is to support all kinds of join type.

But I should keep this check right now, since I have not support some kinds of join.